kms/yubikey/yubikey.go

```diff
@@ -353,10 +354,20 @@ func (k *YubiKey) CreateAttestation(req *apiv1.CreateAttestationRequest) (*apiv1
 		Certificate:         cert,
 		CertificateChain:    []*x509.Certificate{cert, intermediate},
 		PublicKey:           cert.PublicKey,
-		PermanentIdentifier: getSerialNumber(cert),
+		PermanentIdentifier: getAttestedSerial(cert),
 	}, nil
 }
 
+// Serial returns the serial number of the PIV card or and empty
+// string if retrieval fails
+func (k *YubiKey) Serial() string {
+	if serial, err := k.yk.Serial(); err == nil {
+		return strconv.FormatUint(uint64(serial), 10)
+	}
+
+	return ""
+}
```

Comment:
You should return an error, wrapping it with something like:

```go
fmt.Errorf("error getting Yubikey's serial number: %w", err)
```

Reply:
ea7114e